composer/models/huggingface.py

```diff
@@ -296,7 +303,10 @@ def get_metrics(self, is_train: bool = False) -> Dict[str, Metric]:
         return metrics if metrics else {}
 
     def update_metric(self, batch: Any, outputs: Any, metric: Metric) -> None:
-        metric.update(outputs, self.labels)
+        if 'update_metric_handle' in batch:
+            batch['update_metric_handle'](metric, batch, outputs, self.labels)
+        else:
+            metric.update(outputs, self.labels)
```

Comment:
I think the `update_metric_handle` residing in the dataset is slightly against the design principle... could we not just check on `isinstance(metric, InContextLearningLMAccuracy)` and if so, pass in the `(batch, outputs, self.labels)`? The input args are a property of the _metric_, not the _dataset_, whose batches could be fed into many different metrics.

Also, I'm slightly confused why `self.labels` is a thing...  Shouldn't the labels come directly from the batch? Or is this using some HF model thing? Bit of a weird thing to see a `self` property on the model that is changing with each batch O.o

Reply:
Agree with the first suggestion to check the type here rather than stashing the function reference on the batch.

I was also confused by the `self.labels` thing when I looked at this code. It is on my todo list to look at it and try to figure out why it was originally written this way and if it is necessary.